polaris-react/vite.config.mjs

```diff
@@ -0,0 +1,30 @@
+// Vite is used by Storybook.
+// Rollup is used directly to build for prod.
+import {readFileSync} from 'fs';
+
+import {defineConfig} from 'vite';
+
+import {generateScopedName} from './config/rollup/namespaced-classname-module.mjs';
```

Comment:
I had to revisit this one because I'd forgotten the reason we'd gone this route. Turns out it's unfortunately not as simple as node's file type detection; Vite is baulking on the `require('path')` within `config/rollup/namespaced-classname.js`:

```
ERR! Error: Dynamic require of "node:path" is not supported
ERR!     at file:///Users/jess/dev/polaris/polaris-react/vite.config.mjs.timestamp-1682311621838.mjs:12:9
ERR!     at config/rollup/namespaced-classname.js (/Users/jess/dev/polaris/polaris-react/config/rollup/namespaced-classname.js:1:369)
ERR!     at __require2 (file:///Users/jess/dev/polaris/polaris-react/vite.config.mjs.timestamp-1682311621838.mjs:15:50)
ERR!     at <anonymous> (/Users/jess/dev/polaris/polaris-react/vite.config.mjs:7:34)
ERR!     at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
ERR!     at async Promise.all (index 0)
ERR!     at async ESMLoader.import (node:internal/modules/esm/loader:533:24)
ERR!     at async loadConfigFromBundledFile (file:///Users/jess/dev/polaris/node_modules/vite/dist/node/chunks/dep-d305c21f.js:63967:21)
ERR!     at async Module.loadConfigFromFile (file:///Users/jess/dev/polaris/node_modules/vite/dist/node/chunks/dep-d305c21f.js:63852:28)
ERR!     at commonConfig (/Users/jess/dev/polaris/node_modules/@storybook/builder-vite/dist/index.js:161:5305)
```

It looks like Storybook is doing some kind of pre-processing on the files (see the `.timestamp-XXXXX` suffixes) which might be messing with it 🤔